Bug in `uncompress` from `scilpy/tractograms/uncompress.pyx`

There is a bug in the `uncompress` function where more `points_to_idx` may be returned than there are points in the streamline.

Here is an example script to test

```python
import os

from scilpy.io.streamlines import load_tractogram
from scilpy.tractograms.uncompress import uncompress

# Tests should be run locally so that you have the testing data loaded in your home directory
home = os.path.expanduser('~')
sft = load_tractogram(f'{home}/.scilpy/filtering/bundle_all_1mm.trk', 'same')

sft.to_vox()
sft.to_corner()

indices, points_to_idx = uncompress(sft.streamlines, return_mapping=True)
streamline_lengths, mapping_lengths = [len(s) for s in sft.streamlines], [len(m) for m in points_to_idx]

for s,m in zip(streamline_lengths, mapping_lengths):
    assert s == m, (s,m)
```

Should give you an error.

`AssertionError: (130, 131)`

--- a/scilpy/tractograms/uncompress.pyx
+++ b/scilpy/tractograms/uncompress.pyx
@@ -239,36 +239,16 @@ cdef cnp.npy_intp _uncompress(
 
             direction_norm = norm(direction[0], direction[1], direction[2])
 
-            # Make sure that the next point is not exactly on a voxel
-            # intersection or on the face of the voxel, since the behavior
-            # is not easy to define in this case.
-            if fabs(next_pt[0] - floor(next_pt[0])) < 1e-8 or\
-                fabs(next_pt[1] - floor(next_pt[1])) < 1e-8 or\
-                fabs(next_pt[2] - floor(next_pt[2])) < 1e-8:
-                # TODO in future, jitter edge or add thickness to deal with
-                # case where on an edge / face and the corresponding
-                # component of the direction is 0
-                next_pt[0] = next_pt[0] - 0.000001 * direction[0]
-                next_pt[1] = next_pt[1] - 0.000001 * direction[1]
-                next_pt[2] = next_pt[2] - 0.000001 * direction[2]
-
-                # Make sure we don't "underflow" the grid
-                if next_pt[0] < 0.0 or next_pt[1] < 0.0 or next_pt[2] < 0.0:
-                    next_pt[0] = next_pt[0] + 0.000002 * direction[0]
-                    next_pt[1] = next_pt[1] + 0.000002 * direction[1]
-                    next_pt[2] = next_pt[2] + 0.000002 * direction[2]
-
-                # Keep it in mind to correctly set when going back in the loop
-                jittered_point[0] = next_pt[0]
-                jittered_point[1] = next_pt[1]
-                jittered_point[2] = next_pt[2]
-
-                # Update those
-                direction[0] = next_pt[0] - current_pt[0]
-                direction[1] = next_pt[1] - current_pt[1]
-                direction[2] = next_pt[2] - current_pt[2]
-
-                direction_norm = norm(direction[0], direction[1], direction[2])
+            # Shift coordiates if the are too close to the voxel boundaries
+            for i in range(3):
+                if fabs(next_pt[i] - floor(next_pt[i])) < 1e-8:
+                    if next_pt[i] > 0.000001:
+                        next_pt[i] = next_pt[i] - 0.000001
+                    else:
+                        next_pt[i] = next_pt[i] + 0.000001
+                    jittered_point[i] = next_pt[i]
+                    direction[i] = next_pt[i] - current_pt[i]
+                    direction_norm = norm(direction[0], direction[1], direction[2])
 
             # Set the "remaining_distance" var to compute remaining length of
             # vector to process
@@ -295,7 +275,7 @@ cdef cnp.npy_intp _uncompress(
                 # Check if last point is already on an edge
                 remaining_distance -= length_ratio * direction_norm
 
-                if remaining_distance < 0.:
+                if remaining_distance <= 0.:
                     pointers.points_to_index_out[0] = nb_points_out - 1
                     pointers.points_to_index_out += 1
                     nb_points_out_pti += 1

